### PR TITLE
[SPARK-22676][FOLLOW-UP] fix code style for test.

### PR DESCRIPTION
**Pull request by jinxing64**
## What changes were proposed in this pull request?

This pr address comments in https://github.com/apache/spark/pull/19868 ;
Fix the code style for `org.apache.spark.sql.hive.QueryPartitionSuite` by using:
`withTempView`, `withTempDir`, `withTable`...

### CODE REVIEW

**Comment by jinxing64** [comment]
Jenkins, test this please.

**Comment by jinxing64** [comment]
cc @cloud-fan

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala [hunk 1]
```diff
@@ -34,79 +34,83 @@ class QueryPartitionSuite extends QueryTest with SQLTestUtils with TestHiveSingl
...
-        .find { f => f.isDirectory && f.getName().startsWith("ds=") }
-        .foreach { f => Utils.deleteRecursively(f) }
-
-      // test for after delete the path
-      checkAnswer(sql("select key,value from table_with_partition"),
-        testData.toDF.collect ++ testData.toDF.collect ++ testData.toDF.collect)
-
-      sql("DROP TABLE IF EXISTS table_with_partition")
-      sql("DROP TABLE IF EXISTS createAndInsertTest")
+    withSQLConf((SQLConf.HIVE_VERIFY_PARTITION_PATH.key -> "true")) {
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala]: we only need one pair of `()`

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala [hunk 2]
```diff
@@ -34,79 +34,83 @@ class QueryPartitionSuite extends QueryTest with SQLTestUtils with TestHiveSingl
...
+      withTempView("testData") {
+        withTable("table_with_partition", "createAndInsertTest") {
+          withTempPath { tmpDir =>
+            tmpDir.mkdir()
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala]: `withTempDir`, then you don't need to call `mkdir` here

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala [hunk 3]
```diff
@@ -34,79 +34,83 @@ class QueryPartitionSuite extends QueryTest with SQLTestUtils with TestHiveSingl
...
+            sql("INSERT OVERWRITE TABLE table_with_partition  partition (ds='2') " +
+                "SELECT key,value FROM testData")
+            sql("INSERT OVERWRITE TABLE table_with_partition  partition (ds='3') " +
+                "SELECT key,value FROM testData")
+            sql("INSERT OVERWRITE TABLE table_with_partition  partition (ds='4') " +
+                "SELECT key,value FROM testData")
+
+            // test for the exist path
+            checkAnswer(sql("select key,value from table_with_partition"),
+              testData.union(testData).union(testData).union(testData).collect)
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala]: no need to call `collect`, `checkAnswer` can compare 2 dataframes

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala [hunk 4]
```diff
@@ -34,79 +34,81 @@ class QueryPartitionSuite extends QueryTest with SQLTestUtils with TestHiveSingl
...
+            // test for after delete the path
+            checkAnswer(sql("select key,value from table_with_partition"),
+              testData.union(testData).union(testData))
+          }
+        }
+      }
     }
   }
 
   test("Replace spark.sql.hive.verifyPartitionPath by spark.files.ignoreMissingFiles") {
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala]: These 2 tests are exactly same except the config, can we create a common method for it?

Comment (jinxing64) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala]: I extracted a common method, check again.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by jinxing64** [comment]
Thanks for merging !

**Pull request closed without merging**